<launch>
  <arg name="save_point_cloud" doc="Whether to save point clouds" default="true" />
  <arg name="save_images" doc="Whether to save images grouped by image synchroniser" default="true" />
  <arg name="save_header" doc="Header appended to start of saved files" default="log" />
  <arg name="base_frame_id" doc="If specified, will transform point cloud to a particular frame in tf" default="" />

  <node pkg="nodelet" type="nodelet" name="point_cloud_logger" args="load sl_sensor_logger/PointCloudLoggerNodelet standalone_nodelet" output="screen">
    <param name="service_name" type="str" value="/enable_log_pc" />
    <param name="log_topic" type="str" value="/triangulator/point_cloud" />
    <param name="initially_enabled" type="bool" value="$(arg save_point_cloud)" />
    <param name="log_directory" type="str" value="$(find sl_sensor)/saved_point_clouds/" />
    <param name="file_header" type="str" value="$(arg save_header)" />
    <param name="base_frame_id" type="str" value="$(arg base_frame_id)" />
  </node>		  

  <node pkg="nodelet" type="nodelet" name="image_logger" args="load sl_sensor_logger/ImageArrayLoggerNodelet standalone_nodelet" output="screen">
    <param name="service_name" type="str" value="/enable_log_image" />
    <param name="log_topic" type="str" value="/image_synchroniser/grouped_images" />
    <param name="initially_enabled" type="bool" value="$(arg save_images)" />
    <param name="log_directory" type="str" value="$(find sl_sensor)/saved_images/" />
    <param name="file_header" type="str" value="$(arg save_header)" />
  </node>
</launch>